table/tables/index.go

```diff
@@ -130,6 +130,26 @@ func (c *index) GenIndexKey(indexedValues []types.Datum, h int64) (key []byte, d
 			}
 		}
 	}
+
+	if len(indexedValues) > 0 {
+		if len(c.idxInfo.Columns) != len(indexedValues) {
+			err = errors.Errorf("len(indexedValues) should equal to index columns count")
```

Comment:
this is an assertion, not usual mysql errors, caller should guarantee the input's validation.
I can use `panic` instead, but panic is not user friendly.